lib/mirage.ml

```diff
@@ -628,22 +627,20 @@ let opt_map f = function Some x -> Some (f x) | None -> None
 let (@?) x l = match x with Some s -> s :: l | None -> l
 let (@??) x y = opt_map Key.abstract x @? y
 
-let ipv4_keyed_conf ?address ?network ?gateway () = impl @@ object
+let ipv4_keyed_conf ?network ?gateway () = impl @@ object
     inherit base_configurable
     method ty = ethernet @-> arpv4 @-> ipv4
     method name = Name.create "ipv4" ~prefix:"ipv4"
     method module_name = "Static_ipv4.Make"
     method packages = Key.pure ["tcpip"]
     method libraries = Key.pure ["tcpip.ipv4"]
-    method keys = address @?? network @?? gateway @?? []
+    method keys = network @?? gateway @?? []
     method connect _ modname = function
     | [ etif ; arp ] ->
-
         Fmt.strf
-          "%s.connect@[@ %a@ %a@ %a@ %s@ %s@]"
+          "let (ip, network) = Key_gen.network () in @ \
```

Comment:
Do exactly what you did (`let (ip, network) ...`) just use `opt_key` and `%a` instead of hardcoding `Key_gen.network ()`.